### PR TITLE
graph: panic if the graph is corrupted

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-orientdb

File: graffiti/graph/graph.go [hunk 1]
```diff
@@ -1003,9 +1019,11 @@ func (g *Graph) Unlink(n1 *Node, n2 *Node) {
 
 		parent, child := parents[0], children[0]
 		if child.ID == n2.ID || parent.ID == n2.ID {
-			g.DelEdge(e)
+			err = g.DelEdge(e)
```
Comment (lebauce) [inline graffiti/graph/graph.go]: If multiple edges are removed, err could be set at some time and reset later

File: graffiti/graph/indexer.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ func (i *Indexer) Stop() {
 	i.listenerHandler.RemoveEventListener(i)
 }
 
-// AddEventListener subscibe a new graph listener
+// AddEventListener subscribe a new graph listener
```
Comment (lebauce) [inline graffiti/graph/indexer.go]: s/subscribe/subscribes/

File: graffiti/graph/memory.go [hunk 3]
```diff
@@ -41,31 +43,35 @@ type MemoryBackend struct {
 }
 
 // MetadataUpdated returns true
```
Comment (lebauce) [inline graffiti/graph/memory.go]: s/returns true/return nil/

File: graffiti/graph/orientdb.go [hunk 4]
```diff
@@ -88,39 +88,44 @@ func metadataToOrientDBSelectString(m ElementMatcher) string {
...
 	}
 	if len(values.Result) == 0 {
-		return false
+		return ErrElementNotFound
 	}
 
 	value, ok := values.Result[0]["value"]
 	if !ok {
-		return false
+		fmt.Printf("CCCCCCCCCCCCC: %+v\n", values)
```
Comment (lebauce) [inline graffiti/graph/orientdb.go]: nit: remove line

File: graffiti/graph/orientdb.go [hunk 5]
```diff
@@ -88,39 +88,44 @@ func metadataToOrientDBSelectString(m ElementMatcher) string {
...
+		return ErrElementNotFound
 	}
 
 	f, ok := value.(float64)
-	return ok && f == 1
+	if !ok || f != 1 {
+		fmt.Printf("ZZZZZZZZZZZ: %+v\n", values)
```
Comment (lebauce) [inline graffiti/graph/orientdb.go]: nit: remove line

File: graffiti/pod/subscriber_endpoint.go [hunk 6]
```diff
@@ -224,6 +224,10 @@ func (t *TopologySubscriberEndpoint) OnEdgeDeleted(e *graph.Edge) {
 	t.notifyClients(ws.NewStructMessage(graph.Namespace, graph.EdgeDeletedMsgType, e))
 }
 
+// OnError graph error event. Implements the EventListener interface.
+func (t *TopologySubscriberEndpoint) OnError(err error) {
```
Comment (lebauce) [inline graffiti/pod/subscriber_endpoint.go]: This seems to be unused

File: topology/probes/docker/docker.go [hunk 7]
```diff
@@ -145,7 +152,10 @@ func (probe *Probe) unregisterContainer(id string) {
 	}
 
 	probe.Graph.Lock()
-	probe.Graph.DelNode(infos.Node)
+	if err := probe.Graph.DelNode(infos.Node); err != nil {
+		logging.GetLogger().Error(err)
```
Comment (lebauce) [inline topology/probes/docker/docker.go]: You need to unlock the graph before returning

File: topology/probes/ovsdb/ovsdb.go [hunk 8]
```diff
@@ -467,11 +484,18 @@ func (o *Probe) OnOvsPortAdd(monitor *ovsdb.OvsMonitor, uuid string, row *libovs
...
+		var err error
+
+		port, err = o.Graph.NewNode(graph.GenID(), graph.Metadata{
 			"UUID": uuid,
 			"Name": row.New.Fields["name"].(string),
 			"Type": "ovsport",
 		})
+
+		if err != nil {
+			logging.GetLogger().Error(err)
```
Comment (lebauce) [inline topology/probes/ovsdb/ovsdb.go]: We should probably return

**Pull request merged**